### DISCUSSION TITLE
Jetson Nano 2gb Development kit doesn't boot

### DISCUSSION THREAD

**Discussion by ChipHunter** [OE4T/meta-tegra#747] [Q&A] [answered]
Hallo All, 

I am currently working on migrating our application from using Xavier nx to using nano 2gb. Unfortunately I am not able to get the board to boot although I am following the exact same steps as  I did with the xavier nx. The only difference is the machine configuration 

MACHINE = "jetson-nano-2gb-devkit"

I have noticed that the sd card portioning differs greatly between yocto image and the one provided by Nvidia. Please have a look at the below 2 snapshots. The first one is yocto and the other is nvidia. 

mmcblk0     179:0    0  29.7G  0 disk 
└─mmcblk0p1 179:1    0    14G  0 part /media/ammar/91e23521-a787-4974-a0b3-13a5f9d5d840

mmcblk0      179:0    0  29.7G  0 disk 
├─mmcblk0p1  179:1    0  13.3G  0 part /media/ammar/248535ce-9ac0-4041-bcc5-7b5fd94a36bb
├─mmcblk0p2  179:2    0   128K  0 part 
├─mmcblk0p3  179:3    0   448K  0 part 
├─mmcblk0p4  179:4    0   576K  0 part 
├─mmcblk0p5  179:5    0    64K  0 part 
├─mmcblk0p6  179:6    0   192K  0 part 
├─mmcblk0p7  179:7    0   384K  0 part 
├─mmcblk0p8  259:3    0    64K  0 part 
├─mmcblk0p9  259:4    0   448K  0 part 
├─mmcblk0p10 259:5    0   448K  0 part 
├─mmcblk0p11 259:6    0   768K  0 part 
├─mmcblk0p12 259:7    0    64K  0 part 
├─mmcblk0p13 259:8    0   192K  0 part 
└─mmcblk0p14 259:9    0   128K  0 part 

Am I missing something here? 

Thanks in advance!

**Comment by madisongh** [comment]
When you say it doesn't boot, what exactly does that mean - that is, at what point does the boot fail? Do you have a serial console attached, and can you provide a copy of the output there?

Also, which branch are you build from, and did you re-flash the SPI flash before trying to boot?  There was a major change in the boot partition layouts for the Nanos between R32.4.x and R32.5.0, so if your Nano has the R32.5.0 SPI flash layout, it won't boot an image built off an R32.4.x BSP, and vice-versa.

**Comment by ChipHunter** [comment, marked as answer]
Thanks Matt for the quick response! 

I am using the master branch for both poky and meta-tegra. I am also using sd card to boot the board. So after ./dosdcard.sh, I copy the image to the sd card using dd. The same method I used to boot the nx before. I also attached a serial cable and captured the following output: 

00.472] Set High speed to 1
[0000.475] Error mask set in wait for cmd complete with error 0x3 in HwSdmmcWaitForCommandComplete func at 278 line 
[0000.485] Command complete wait failed with error 0x3 Interrupt 0x18001
[0000.492] Number of retries left 4
[0000.501] Using GPT Primary to query partitions
[0000.505] Read PT from (2:0)
[0000.521] Using BFS PT to query partitions 
[0000.525] PT: Partition TBC NOT found ! 
[0000.528] Warning: Find Partition via PT Failed
[0000.533] failed to load NvTbootTbootCpu from (4:0)
[0000.537] re-load NvTbootTbootCpu from (2:0)
[0000.541] Read GPT from (4:0)
[0000.549] Using GPT Primary to query partitions
[0000.554] PT: Partition TBC NOT found ! 
[0000.558] Warning: Find Partition via PT Failed
[0000.562] Read GPT from (4:0)
[0000.570] Using GPT Primary to query partitions
[0000.575] PT: Partition TBC NOT found ! 
[0000.579] Warning: Find Partition via PT Failed
[0000.583] Error is 1 

It seems that some partitions are missing and I don't know why. I am not really sure what you mean by "and did you re-flash the SPI flash before trying to boot?". Could you please elaborate more?

**Reply by madisongh** [reply]
I've added some notes [here](https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit#spi-flash-on-sdcard-based-kits) on the wiki about this.

**Reply by ChipHunter** [reply]
Thank you very much! it is booting now.
